src/test/java/org/broadinstitute/hellbender/tools/walkers/vqsr/VariantRecalibratorIntegrationTest.java

```diff
@@ -0,0 +1,154 @@
+                    getLargeVQSRTestDataDir() + "phase1.projectConsensus.chr20.1M-10M.raw.snps.vcf",
+                    "-L","20:1,000,000-10,000,000",
+                    "--resource",
+                    "known,known=true,prior=10.0:" + getLargeVQSRTestDataDir() + "dbsnp_132_b37.leftAligned.20.1M-10M.vcf",
+                    "--resource",
+                    "truth_training1,truth=true,training=true,prior=15.0:" + getLargeVQSRTestDataDir() + "sites_r27_nr.b37_fwd.20.1M-10M.vcf",
+                    "--resource",
+                    "truth_training2,training=true,truth=true,prior=12.0:" + getLargeVQSRTestDataDir() + "Omni25_sites_1525_samples.b37.20.1M-10M.vcf",
+                    "-an", "QD", "-an", "HaplotypeScore", "-an", "HRun",
+                    "--trustAllPolymorphic", // for speed
+                    "-mode", "SNP"
+                }
+            },
+        };
+    }
+
+    @Test(dataProvider = "VarRecalSNP")
+    public void testVariantRecalibratorSNP(final String[] params) throws IOException {
+        //NOTE: The number of iterations required to ensure we have enough negative training data to proceed,
+        //as well as the test results themselves, are both very sensitive to the state of the random number
+        //generator at the time the tool starts to execute. Sampling a single integer from the RNG at the
+        //start aligns the initial state of the random number generator with the initial state of the GATK3
+        //random number generator at the time the tests start executing (both RNGs use the same seed, but
+        //the pathway to the test in GATK3 results in an additional integer being sampled), thus allowing the
+        //results to be easily compared against those produced by GATK3. This also happens to allow this
+        //test to succeed on the first iteration (max_attempts=1) which we want for test performance reasons.
+        //Failing to call nextInt here would result in the model failing on the first 3 attempts (it would
+        //succeed on the 4th), and the results would not match those produced by GATK3 on these same inputs.
+        //
+        //Also note that due to this RNG conditioning, this test will produce different results when manually
+        //outside of the test framework.
+        @SuppressWarnings("unused")
+        final int hack = Utils.getRandomGenerator().nextInt();
+
+        // use an ArrayList - ArgumentBuilder tokenizes using the "=" in the resource args
+        List<String> args = new ArrayList<>(params.length);
+        Stream.of(params).forEach(arg -> args.add(arg));
+
+        File recalOut = createTempFile("testVarRecalSnp", ".vcf");
+        File tranchesOut = createTempFile("testVarRecalSnp", ".txt");
+        args.addAll(addTempFileArgs(recalOut, tranchesOut));
+
+        final VariantRecalibrator varRecalTool = new VariantRecalibrator();
+        Assert.assertEquals(varRecalTool.instanceMain(args.toArray(new String[args.size()])), true);
```

Comment:
Is there a reason you're invoking the tool this way instead of calling `runCommandLine()`?